pkgpanda/docker/dcos-builder/Dockerfile

```diff
@@ -1,39 +1,46 @@
-FROM ubuntu:14.04.4
+FROM ubuntu:16.04
 MAINTAINER help@dcos.io
 
 RUN apt-get -qq update && apt-get -y install \
   autoconf \
   automake \
   cmake \
-  make \
-  gcc \
   cpp \
-  patch \
-  python-dev \
-  python-pip \
-  git \
-  libtool \
+  curl \
   default-jdk \
   default-jre \
+  dpkg-dev \
+  g++-4.8 \
+  gcc-4.8 \
+  gettext-base \
+  git \
   gzip \
-  zlib1g-dev \
+  libapr1-dev \
   libcurl4-openssl-dev \
-  python-setuptools \
-  dpkg-dev \
+  libnl-3-dev \
+  libnl-genl-3-dev \
+  libpcre++-dev \
+  libpopt-dev \
   libsasl2-dev \
-  maven \
-  libapr1-dev \
   libsvn-dev \
+  libtool \
+  linux-headers-4.4.0-45-generic \
+  make \
+  maven \
+  patch \
+  pkg-config \
+  python-dev \
+  python-pip \
+  python-setuptools \
   ruby \
-  curl \
-  wget \
   scala \
+  unzip \
+  wget \
+  xutils-dev \
   xz-utils \
-  libpopt-dev \
-  libnl-3-dev \
-  libnl-genl-3-dev \
-  linux-headers-3.19.0-21-generic \
-  pkg-config \
-  gettext-base \
-  unzip
+  zlib1g-dev
+
+ENV CC=/usr/bin/gcc-4.8
```

Comment:
Sounds good for now. Definitely we're going to need to bundle libstdc++ in the future (And maybe libc as well).